jetty-io/src/main/java/org/eclipse/jetty/io/ManagedSelector.java

```diff
@@ -495,47 +509,46 @@ public String toString()
 
     private class DumpKeys extends Invocable.NonBlocking
     {
-        private final CountDownLatch latch = new CountDownLatch(1);
-        private final List<Object> _dumps;
-
-        private DumpKeys(List<Object> dumps)
-        {
-            this._dumps = dumps;
-        }
-
+        private CountDownLatch latch = new CountDownLatch(1);
+        private List<String> keys;
+        
         @Override
         public void run()
         {
             Selector selector = _selector;
+            List<String> list = new ArrayList<>(selector.keys().size()+1);
```

Comment:
This is not right, since `selector` may be null and will produce a NPE here. Move it inside the `if` statement below, and you will also already have the `keys` set.